# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/WidgetEventImpl.cpp
# Commit: 5718f7f3067e
# Full Hash: 5718f7f3067eff42e097a6013acd82fbe6aa299a
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2025-05-15 22:58:00
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar
#   
#   This patch tries to dispatch `ePointerRawUpdate` with
#   `PresShell::EventHandler::DispatchPrecedingPointerEvent` as same as usual
#   pointer events.
# ==============================================================================

diff -r aae164cff6aa -r 5718f7f3067e widget/WidgetEventImpl.cpp
--- a/widget/WidgetEventImpl.cpp	Thu May 15 03:20:28 2025 +0000
+++ b/widget/WidgetEventImpl.cpp	Thu May 15 03:20:29 2025 +0000
@@ -174,6 +174,12 @@
     case eTouchPointerCancel:
       return true;
 
+    case eMouseRawUpdate:
+    case eTouchRawUpdate:
+      MOZ_ASSERT_UNREACHABLE(
+          "Internal raw update events shouldn't be dispatched to the DOM");
+      return true;
+
     default:
       return false;
   }
@@ -384,6 +390,7 @@
     case eMouseOut:
     case eMouseHitTest:
     case eMouseMove:
+    case eMouseRawUpdate:
       return true;
     // TODO: Perhaps, we should rename this method.
     case ePointerClick:
@@ -531,7 +538,7 @@
 bool WidgetEvent::IsAllowedToDispatchDOMEvent() const {
   switch (mClass) {
     case eMouseEventClass:
-      if (mMessage == eMouseTouchDrag) {
+      if (mMessage == eMouseRawUpdate || mMessage == eMouseTouchDrag) {
         return false;
       }
       [[fallthrough]];
@@ -541,7 +548,7 @@
       // DOM events.
       // Synthesized button up events also do not cause DOM events because they
       // do not have a reliable mRefPoint.
-      return AsMouseEvent()->mReason == WidgetMouseEvent::eReal;
+      return AsMouseEvent()->IsReal();
 
     case eWheelEventClass: {
       // wheel event whose all delta values are zero by user pref applied, it
@@ -551,7 +558,7 @@
              wheelEvent->mDeltaZ != 0.0;
     }
     case eTouchEventClass:
-      return mMessage != eTouchPointerCancel;
+      return mMessage != eTouchRawUpdate && mMessage != eTouchPointerCancel;
     // Following events are handled in EventStateManager, so, we don't need to
     // dispatch DOM event for them into the DOM tree.
     case eQueryContentEventClass:
@@ -908,6 +915,7 @@
   switch (mMessage) {
     // This method is designed for mouse events.
     case eMouseMove:
+    case eMouseRawUpdate:
     case eMouseUp:
     case eMouseDown:
     case eMouseEnterIntoWidget:
@@ -936,6 +944,7 @@
     case ePointerClick:
     case ePointerAuxClick:
     case ePointerMove:
+    case ePointerRawUpdate:
     case ePointerUp:
     case ePointerDown:
     case ePointerCancel:
